Assets/MixedRealityToolkit/_Core/Definitions/SpatialAwareness/MixedRealitySpatialAwarenessProfile.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Internal.Definitions.SpatialAwareness
+{
+    /// <summary>
+    /// Configuration profile settings for setting up spatial awareness.
+    /// </summary>
+    [CreateAssetMenu(menuName = "Mixed Reality Toolkit/Mixed Reality Spatial Awareness Profile", fileName = "MixedRealitySpatialAwarenessProfile", order = 7)]
+    public class MixedRealitySpatialAwarenessProfile : ScriptableObject
+    {
+        [SerializeField]
+        private bool autoStart = true;
```

Comment:
Because you may actually want to defer observation startup to a specific point in the experience. If we disable the system in the config profile, we lose the dev's desired spatial profile.

Reply:
>We don't need auto start.

Why not? An app experience might want the system set up but doesn't want it to start mapping or analyzing until a specific trigger. On the other hand, some apps will want it all started up at launch.

Reply:
>Why not? An app experience might want the system set up but doesn't want it to start mapping or analyzing until a specific trigger. On the other hand, some apps will want it all started up at launch.

Because it would be up to the developer as to _when_ they choose to enable it. Similar to how the boundary system can enable/disable it's visualizations.

It could start disabled, then when the app reaches a point where it needs to be on, it enables it. No auto anything should be in the core system. That's up to how the developer implements their app, not how we choose to setup the system.

Reply:
Either it starts enabled or it doesn't. Simple as that.

Reply:
>Because it would be up to the developer as to when they choose to enable it.

Yes, that's why this is an option in the profile. The developer gets to choose their profile settings.

>Either it starts enabled or it doesn't. Simple as that.

How is this different from that? The system is set up, and, according to this setting, mapping is either enabled or isn't.

Reply:
> Yes, that's why this is an option in the profile. The developer gets to choose their profile settings.

The difference is that it's redundant bc you just don't need to enable system. Then in code the dev can specify when to enable it.

Reply:
It's not like I'm saying we shouldn't delay starting the system until later. Far from it. 

I'm saying that this bool is redundant, because for a system to auto start it needs to be enabled and registered with the MRManager, but then we're gonna enable it when it starts, then just disable it during that startup phase? That's unnecessary. Disable it, but keep the ability to setup the system in the config profile, and let the user turn it on when they feel like they need it.

Reply:
I intended to name this autoStartObserver (like in the HTK component) and treat it like the boundary system's showPlayArea, etc.

Essentially, the system is up and running, but we defer the actual underlying platform behavior until the application desires it. I recommend taking a look at Holograms 230 in the Academy content for an example of autoStartObserver.

FYI: we need to have a way for users to control spatial awareness as there is a performance cost associated.

Look out for a detailed architecture proposal doc "soon".